jeremyhager/argo-workflows
# feat: Add Rust SDK

### Motivation
The SDK for Argo Workflows provides 2 official packages: Python and Java. However, this is lacking for other languages including Rust. It is possible to manually create one, however 

### Modifications

- Add SDK for rust under `sdks/rust`
  - `Makefile` for generating, "installing", and testing
  - `README.md` for getting started
  - `.gitignore` as necessary for files and folders
- Modify `.github/ci-build.yaml`
  - Add test for Rust SDK under `e2e-tests`
  - Add `actions-rs` for Rust `matrix-test`
  - Update `make cli STATIC_FILES=false` with Rust matrix test
  - Update `make start PROFILE` with Rust matrix test
  - Update `Start controller/API` with Rust matrix test
- Modify `.github/sdks.yaml`
  - Update strategy to include `rust`
  - Update to include `CARGO_REGISTRY_TOKEN`
- Modify `Makefile`
  - Add `make --directory sdks/rust generate` to `codegen` command
 
### Verification

- Run k3d local cluster and `make test` rust sdk against local
- Run README.md tests within `sdks/rust` to confirm examples work
- Run `make pre-commit -B`
- Run draft PR to check changes
